File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int md = 0, l = -1;
        vector<int> mp(128);
        for(auto r = 0; r < s.size(); r++) {
            while (mp[s[r]]) {
                l++;
                mp[s[l]]--;
            }
            mp[s[r]]++;
            md = max(md, r - l);
        }
        return md;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.go
func lengthOfLongestSubstring(s string) int {
    maxL := 0
    freq := make([]int, 127)

    l, r := 0, 0
    for r < len(s) {
        ch := s[r]
        freq[ch]++

        for freq[ch] > 1 {
            freq[s[l]]--
            l++
        }

        maxL = max(maxL, r - l + 1)
        r++
    }

    return maxL
}